File: Entities/Surgery.cs
namespace PrivateHospitalSystem.Entities
{
    public enum SurgeryStatus
    {
        Scheduled,
        InProgress,
        Completed,
        Cancelled
    }

    public class Surgery
    {
        public Guid Id { get; set; }

        public Guid PatientId { get; set; }
        public Patient? Patient { get; set; }

        public Guid RoomId { get; set; } 
        public Room? Room { get; set; }

        public string ProcedureName { get; set; } = string.Empty; 
        public DateTime ScheduledAt { get; set; }
        public int DurationMinutes { get; set; } = 60;

        public SurgeryStatus Status { get; set; } = SurgeryStatus.Scheduled;
        public string? Notes { get; set; }

        // Equipa cirúrgica (relação muitos-para-muitos com Doctor)
        public ICollection<SurgeryTeamMember> Team { get; set; } = new List<SurgeryTeamMember>();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/src/types/surgery.ts
export interface Surgery {
  id: string
  patientId: string
  patientName: string
  roomId: string
  roomNumber: string
  procedureName: string
  scheduledAt: string
  durationMinutes: number
  status: string
  notes: string | null
  team: string[]
}

export interface CreateSurgeryDto {
  patientId: string
  roomId: string
  procedureName: string
  scheduledAt: string
  durationMinutes: number
  notes?: string
}

export interface AddSurgeryTeamMemberDto {
  doctorId: string
  role: string
}